grails-app/domain/org/pih/warehouse/order/Order.groovy

```diff
@@ -324,6 +325,10 @@ class Order implements Serializable {
         return orderItems.any { it.hasInvoices } || orderAdjustments.any { it.hasInvoice }
     }
 
+    Boolean getHasPrepaymentInvoice() {
+        return orderItems.any { it.hasPrepaymentInvoice } || orderAdjustments.any { it.hasPrepaymentInvoice }
```

Comment:
This one is going to be a doozy. Let's add a comment stating that this should be used with caution and never used when displaying a list of orders.